baybe/surrogates/gaussian_process/presets/default.py

```diff
@@ -82,24 +148,67 @@ def _default_noise_factory(
 ) -> tuple[GammaPrior, float]:
     """Create the default noise settings for the Gaussian process surrogate.
 
+    This is taking the low and high dimensional limits of
+    :func:`baybe.surrogates.gaussian_process.presets.default._edbo_noise_factory`
+    and interpolates the prior moments linearly between them.
+    """
+    # TODO: Replace this function with a proper likelihood factory
+
+    # Calculate effective dimensions, ignoring task parameters
+    n_task_parameters = len(
+        [p for p in searchspace.parameters if isinstance(p, TaskParameter)]
+    )
+    effective_dims = train_x.shape[-1] - n_task_parameters
+
+    # Fix dimensionality limits
+    LOW_D_LIM, HIGH_D_LIM = 8, 100
+
+    # Apply prior logic
+    if effective_dims < LOW_D_LIM:
+        # Low D prior
+        return [GammaPrior(1.05, 0.5), 0.1]
+    elif effective_dims > HIGH_D_LIM:
+        # High D prior
+        return [GammaPrior(1.5, 0.1), 5.0]
+    else:
+        # Interpolate prior moments linearly between low D and high D regime
+        interp_factor = (effective_dims - LOW_D_LIM) / (HIGH_D_LIM - LOW_D_LIM)
+
+        def _interp(a, b):
+            return a + (b - a) * interp_factor
+
+        return [GammaPrior(_interp(1.05, 1.5), _interp(0.5, 0.1)), _interp(0.1, 5.0)]
+
+
+def _edbo_noise_factory(
+    searchspace: SearchSpace, train_x: Tensor, train_y: Tensor
+) -> tuple[GammaPrior, float]:
+    """Create the default noise settings for the Gaussian process surrogate.
+
     The logic is adapted from EDBO (Experimental Design via Bayesian Optimization).
 
     References:
-    *   https://github.com/b-shields/edbo
-    *   https://doi.org/10.1038/s41586-021-03213-y
+        * https://github.com/b-shields/edbo/blob/master/edbo/bro.py#L664
+        * https://doi.org/10.1038/s41586-021-03213-y
     """
     # TODO: Replace this function with a proper likelihood factory
 
+    # Calculate effective dimensions, ignoring task parameters
```

Comment:
This code is contained 3 times in this file. Small utility?

Reply:
open to that, prob best part of the searchspace?

Reply:
Having a small `@property` for `effective_dims` feels like something that might be nice from both a user view (--> how much parameters do I have effectively?) and obviously also from a coding perspective.

Reply:
this is kind of what it says directly in the line below so let me know if I should delete that because having it in both places feels weird

Reply:
added